app/src/main/java/ch/sdp/vibester/fragment/ScoreBoardFragment.kt

```diff
@@ -26,42 +28,46 @@ import com.google.firebase.database.ValueEventListener
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
-class ScoreBoardActivity : AppCompatActivity(), OnItemClickListener {
+class ScoreBoardFragment : Fragment(), OnItemClickListener, View.OnClickListener{
     private val dbRef: DatabaseReference = Database.get().getReference("users")
     private var players: MutableList<User>? = null
     private var userScoreboardAdapter: UserScoreboardAdapter? = null
     private var genre: String = ""
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.scoreboard_scoreboard)
-        setSupportActionBar(findViewById(R.id.toolbar))
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_scoreboard, container, false)
+        view.findViewById<Button>(R.id.kpopButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.rockButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.btsButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.topTracksButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.imagDragonsButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.billieEilishButton).setOnClickListener(this)
+        return view
```

Comment:
Very minor nitpick but these lines could be refactored in for example a "toggleSetOnClickListener" function that takes as an argument the button's ID and se
`fun toggleOnClickListener(id: Int): {
view.findViewById<Button>(id).setOnClickListener(this)
}`

Reply:
That or you could even create an Array with all the button IDs, and make this "toggleOnClickListener" function into a for loop setting the listener for each button in the array.

Reply:
I dont like this structure either, to be honest. Also we use these button in several cases and they are handled differently. I will work on it next or, thanks!